packages/lesmis/src/components/DashboardSearch/DashboardSearch.js

```diff
@@ -0,0 +1,199 @@
+import InputBase from '@material-ui/core/InputBase';
+import MuiIconButton from '@material-ui/core/IconButton';
+import MuiSearchIcon from '@material-ui/icons/Search';
+import CloseIcon from '@material-ui/icons/Close';
+import { useDashboardData } from '../../api';
+import { usePortal, useUrlParams, useAutocomplete } from '../../utils';
+import { DashboardSearchResults } from './DashboardSearchResults';
+import { SUB_DASHBOARD_OPTIONS } from '../../constants';
+
+// Setting the button size directly allows a smooth animation
+const BUTTON_SIZE = '50px';
+const EXPANDED_BUTTON_SIZE = '90px';
+
+const SearchButton = styled(MuiIconButton)`
+  border: 1px solid ${props => props.theme.palette.grey['400']};
+  border-radius: 3px;
+  height: ${BUTTON_SIZE};
+  width: ${BUTTON_SIZE};
+  transition: height 0.1s ease, width 0.1s ease, margin 0.1s ease;
+
+  .MuiSvgIcon-root {
+    font-size: 1.8rem;
+  }
+
+  &.active {
+    height: ${EXPANDED_BUTTON_SIZE};
+    width: ${EXPANDED_BUTTON_SIZE};
+    margin-left: ${props => props.theme.spacing(-3)}px;
+    background: ${props => props.theme.palette.primary.main};
+    border-radius: 0;
+    border: none;
+    color: white;
+
+    &:hover {
+      background: ${props => props.theme.palette.primary.main};
+    }
+  }
+`;
+
+const SearchContainer = styled(MuiPaper)`
+  display: flex;
+  align-items: center;
+  top: 1px;
+  right: 0;
+  left: 0;
+  bottom: 1px;
+  z-index: 1;
+  border-radius: 0;
+  background: white;
+  padding-right: 15px;
+
+  &.active {
+    position: absolute;
+    padding-left: ${props => props.theme.spacing(3)}px;
+  }
+`;
+
+const Input = styled(InputBase)`
+  width: 0;
+
+  &.active {
+    width: 100%;
+    padding-left: 15px;
+  }
+`;
+
+const ClearButton = styled(MuiIconButton)`
+  align-self: center;
+  width: 50px;
+  height: 50px;
+  background-color: rgba(0, 0, 0, 0.05);
+
+  &:hover {
+    background-color: rgba(0, 0, 0, 0.1);
+  }
+`;
+
+const useDashboardItems = () => {
+  const { entityCode } = useUrlParams();
+
+  const { data } = useDashboardData({
+    entityCode,
+    includeDrillDowns: false,
+  });
+
+  if (!data) {
+    return [];
+  }
+
+  return data.reduce((allItems, { dashboardName, entityName, ...dashboard }) => {
+    const subDashboardName = SUB_DASHBOARD_OPTIONS.find(option =>
+      dashboard.dashboardCode.startsWith(`LESMIS_${option.code}`),
+    )?.label;
+    const items = dashboard.items.map(item => ({
+      ...item,
+      subDashboardName,
+      dashboardName,
+      entityName,
+    }));
+    return [...allItems, ...items];
+  }, []);
+};
+
+const LIMIT = 50;
+
+export const DashboardSearch = ({ onToggleSearch, getResultsEl }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [isActive, setIsActive] = useState(false);
+  const options = useDashboardItems();
+
+  const resetSearch = () => {
+    onToggleSearch(false);
+    setIsActive(false);
+    setInputValue('');
+  };
+
+  const searchResults = useAutocomplete({
```

Comment:
When seeing `searchResults` in later contexts I kept assuming it would be an array of the results, where is this is more of a complex object including prop getters etc. Perhaps we should call it something like `autocompleteResponse` to avoid that confusion

Reply:
Yes, good suggestion!